chi/server.py

```diff
@@ -427,20 +429,43 @@ def show_server_by_name(name) -> NovaServer:
     return show_server(server_id)
 
 
-def associate_floating_ip(server_name, floating_ip_address=None):
-    server_id = get_server_id(server_name)
 
-    if floating_ip_address:
-        fip = get_floating_ip(floating_ip_address)
-    else:
+def associate_floating_ip(server_name, floating_ip_address=None):
+    #TODO: This function should add fip to a port, not a server
+    server = get_server(server_name)
+    
+    if floating_ip_address == None:
         fip = get_free_floating_ip()
-
+    else:
+        fip = get_floating_ip(floating_ip_address)
+  
+    if fip == None:
+        return
     ip = fip['floating_ip_address']
-    connection().compute.add_floating_ip_to_server(server_id, ip)
-
+    
+    # using method from https://github.com/ChameleonCloud/horizon/blob/f5cf987633271518970b24de4439e8c1f343cad9/openstack_dashboard/api/neutron.py#L518
+    ports = neutron().list_ports(**{'device_id': server.id}).get('ports')
+    
+    fip_target = {
+        'port_id': ports[0]['id'],
```

Comment:
This doesn't check that `ports` is empty, nor does it check that `fixed_ips` is entry. Lots that can break here.